testplan/testing/multitest/driver/kafka.py

```diff
@@ -82,8 +82,12 @@ def __init__(
 
     @emphasized
     @property
-    def host(self) -> Optional[str]:
+    def host(self) -> str:
         """Host to bind to."""
+        if self._host is None:
+            raise RuntimeError(
+                "Host not resolved yet, shouldn't be accessed now."
+            )
```

Comment:
Hm, cannot we just return the None? I think this will immediately break at least metadata extraction unless the exception handling is there. I would suggest making log dependent if that is why we need it.

Reply:
but I think metadata extraction will only be triggered after driver has at least finished with pre_start step? otherwise there won't be any metadata to be extracted I think

Reply:
It works with ContextValue as well. But it is true we are not using it other than Proxima after_start.